platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```
@@ -0,0 +1,216 @@
+    this.factory = factory;
+    this.scheduler =
+        Executors.newSingleThreadScheduledExecutor(
+            StandardThreadFactoryBuilder.newThreadFactory("crlThread"));
+  }
+
+  /** Pulls down the CRL file provided via URL and writes it locally. */
+  @Override
+  public synchronized void run() {
+
+    if (crlByUrlEnabled && crlLocationUrl != null && crlLocationUrl.startsWith(HTTPS)) {
+
+      try {
+        SecureCxfClientFactory cxfClientFactory =
+            factory.getSecureCxfClientFactory(crlLocationUrl, WebClient.class);
+        WebClient client = cxfClientFactory.getWebClient();
+        Response response = client.get();
+
+        Object entity = response.getEntity();
+        if (entity instanceof InputStream) {
+          try (ByteArrayInputStream inputStream =
+              IOUtils.loadIntoBAIS((InputStream) response.getEntity())) {
+
+            byte[] buffer = new byte[inputStream.available()];
+            int read = inputStream.read(buffer);
+
+            if (read > 0) {
+              File crlFile;
+              String localCrlPath;
+
+              if (Strings.fromByteArray(buffer).contains("-----BEGIN")) {
+                localCrlPath = crlFileLocation + PEM_CRL;
+                crlFile = new File(crlFileLocation + PEM_CRL);
+              } else {
+                localCrlPath = crlFileLocation + DEM_CRL;
+                crlFile = new File(localCrlPath);
+              }
+
+              try (OutputStream outStream = new FileOutputStream(crlFile)) {
+                outStream.write(buffer);
+                SecurityLogger.audit(
+                    "Copied the content of the CRl at {} to the local CRL at {}.",
+                    crlLocationUrl,
+                    localCrlPath);
+
+                setCrlFileLocationInPropertiesFile(localCrlPath);
+              }
+            }
+          }
+        }
+      } catch (IOException e) {
+        LOGGER.warn("Unable to copy CRL to local CRL.");
```

❓ Should we include a little more diagnostic information, since we're swallowing the exception? Maybe include `e.getMessage()`.